monitorcontrol/monitorcontrol.py

```diff
@@ -135,6 +159,160 @@ def _get_vcp_feature(self, code: vcp.VCPCode) -> int:
         current, maximum = self.vcp.get_vcp_feature(code.value)
         return current
 
+    def _extract_a_cap(self, caps_str: str, key: str) -> str:
+        """
+        Splits the capabilities string into individual sets.
+
+        Returns:
+            Dict of all values for the capability
+        """
+        start_of_filter = caps_str.upper().find(key.upper())
+
+        if start_of_filter == -1:
+            # not all keys are returned by monitor.
+            # Also sometimes the string has errors.
+            return ""
+
+        start_of_filter += len(key)
+        filtered_caps_str = caps_str[start_of_filter:]
+        end_of_filter = 0
+        for i in range(len(filtered_caps_str)):
+            if filtered_caps_str[i] == "(":
+                end_of_filter += 1
+            if filtered_caps_str[i] == ")":
+                end_of_filter -= 1
+            if end_of_filter == 0:
+                # dont change end_of_filter to remove the closing ")"
+                break
+
+        # 1:i to remove the first character "("
+        return filtered_caps_str[1:i]
+
+    def _convert_to_dict(self, caps_str: str) -> dict:
```

Comment:
This does not use anything from `self`, would be good to move outside of the class, makes unit testing easier.